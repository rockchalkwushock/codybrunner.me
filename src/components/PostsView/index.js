import React from 'react'

import { FlexContainer, Grid, List } from '../commons'
import Card from '../Card'

const PostsView = ({ posts }) => (
  <FlexContainer>
    <Grid alternate>
      <h1>Cody's Blog</h1>
      <List>
        {posts.map(({ node: post }) => (
          <Card key={post.fields.slug} post={post} />
        ))}
      </List>
    </Grid>
  </FlexContainer>
)

export default PostsView
