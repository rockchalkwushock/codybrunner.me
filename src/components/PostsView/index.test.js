import React from 'react'
import { shallow } from 'enzyme'

import PostsView from './'

const { siteMetadata } = require('../../../gatsby-config')

const template = {
  node: {
    excerpt: 'blah blah blah',
    fields: {
      slug: '/turdsspace/'
    },
    frontmatter: {
      date: 'February 03, 2018',
      draft: false,
      tags: ['JavaScript', 'fp', 'oop'],
      title: 'All About JS'
    }
  }
}

const ctx = {
  page: 1,
  pages: 2,
  nodes: new Array(15).fill(template)
}

describe('Component: <PostsView />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<PostsView ctx={ctx} site={siteMetadata} />)
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should have 1 post in list', () => {
    expect(wrapper.find('List').children()).toHaveLength(15)
  })
})
