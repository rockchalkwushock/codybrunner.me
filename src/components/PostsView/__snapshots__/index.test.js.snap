// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <PostsView /> should render correctly 1`] = `
<FlexContainer>
  <SEO
    site={
      Object {
        "aboutSnippet": "Cody Brunner is a full-stack JavaScript developer & Navy Veteran residing in Wichita, Kansas. Cody primarily works with Node, React, & GraphQL. When not writing code he loves to go hiking, play with his dog & niece, and watchhis Jayhawks win!",
        "author": "Cody Brunner",
        "business": Array [
          Object {
            "className": "fas, envelope",
            "href": "mailto:rockchalkwushock@icloud.com",
            "label": "Email",
          },
          Object {
            "className": "fab, telegram-plane",
            "href": "https://t.me/rockchalkwushock",
            "label": "Telegram",
          },
          Object {
            "className": "fas, file-pdf",
            "href": "https://www.dropbox.com/s/hokjljqc8iob7xd/Cody%20A%20Brunner%20-%20Web%20Developer%20Resume.pdf?dl=1",
            "label": "Resume",
          },
        ],
        "contacts": Array [
          Object {
            "className": "fab, github",
            "href": "https://github.com/rockchalkwushock",
            "label": "Github",
          },
          Object {
            "className": "fab, instagram",
            "href": "https://www.instagram.com/rockchalkwushock",
            "label": "Instagram",
          },
          Object {
            "className": "fab, linkedin",
            "href": "https://www.linkedin.com/in/cody-brunner",
            "label": "LinkedIn",
          },
          Object {
            "className": "fas, rss",
            "href": "rss.xml",
            "label": "RSS Feed",
          },
          Object {
            "className": "fab, twitter",
            "href": "https://twitter.com/RockChalkDev",
            "label": "Twitter",
          },
          Object {
            "className": "fab, youtube",
            "href": "https://www.youtube.com/channel/UCZgBTMhX7jZTkbm7Fpv2bWw",
            "label": "Youtube",
          },
        ],
        "copyright": "Â© 2017-2018 Cody Brunner",
        "description": "Cody Brunner is a Full-Stack JavaScript Developer living in Wichita, Kansas",
        "disqusShortname": "codybrunner-rocks",
        "employment": "Are you interested in contacting Cody for employment opportunities? Feel free to reach out to him via email, Telegram, or go ahead and checkout his resume hosted on DropBox.",
        "googleVerify": "TODO",
        "jobTitle": "Full-Stack JavaScript Developer",
        "keywords": "Cody Brunner, Full-Stack JavaScript Developer, Wichita, web-development, tech blog, Node.js, React.js, Apollo/GraphQL",
        "lang": "en_US",
        "links": Object {
          "creativeCommons": Object {
            "href": "https://creativecommons.org/licenses/by-nc-sa/4.0/",
            "text": "All Rights Reserved.",
          },
          "gatsby": Object {
            "href": "https://gatsbyjs.org",
            "text": "Powered by Gatsby 1.9",
          },
          "now": Object {
            "href": "https://zeit.co/now",
            "text": "Hosted on Now",
          },
          "src": Object {
            "href": "https://github.com/rockchalkwushock/codybrunner.rocks",
          },
          "styled": Object {
            "href": "https://www.styled-components.com",
            "text": "Styled with Styled-Components 3.1",
          },
        },
        "menu": Array [
          Object {
            "href": "/",
            "id": 1,
            "text": "Home",
          },
          Object {
            "href": "/about",
            "id": 2,
            "text": "About",
          },
          Object {
            "href": "/posts/1",
            "id": 3,
            "text": "Blog",
          },
        ],
        "siteUrl": "/",
        "title": "Cody Brunner - Full-Stack JavaScript Developer",
        "twitter": "@RockChalkDev",
      }
    }
  />
  <Grid
    alternate={true}
  >
    <h1>
      Cody's Blog
    </h1>
    <List
      flow="row"
      gap={20}
    >
      <SiteCard
        key="/turdsspace/"
        post={
          Object {
            "excerpt": "blah blah blah",
            "fields": Object {
              "slug": "/turdsspace/",
            },
            "frontmatter": Object {
              "date": "February 03, 2018",
              "draft": false,
              "tags": Array [
                "JavaScript",
                "fp",
                "oop",
              ],
              "title": "All About JS",
            },
          }
        }
      />
      <SiteCard
        key="/turdsspace/"
        post={
          Object {
            "excerpt": "blah blah blah",
            "fields": Object {
              "slug": "/turdsspace/",
            },
            "frontmatter": Object {
              "date": "February 03, 2018",
              "draft": false,
              "tags": Array [
                "JavaScript",
                "fp",
                "oop",
              ],
              "title": "All About JS",
            },
          }
        }
      />
      <SiteCard
        key="/turdsspace/"
        post={
          Object {
            "excerpt": "blah blah blah",
            "fields": Object {
              "slug": "/turdsspace/",
            },
            "frontmatter": Object {
              "date": "February 03, 2018",
              "draft": false,
              "tags": Array [
                "JavaScript",
                "fp",
                "oop",
              ],
              "title": "All About JS",
            },
          }
        }
      />
      <SiteCard
        key="/turdsspace/"
        post={
          Object {
            "excerpt": "blah blah blah",
            "fields": Object {
              "slug": "/turdsspace/",
            },
            "frontmatter": Object {
              "date": "February 03, 2018",
              "draft": false,
              "tags": Array [
                "JavaScript",
                "fp",
                "oop",
              ],
              "title": "All About JS",
            },
          }
        }
      />
      <SiteCard
        key="/turdsspace/"
        post={
          Object {
            "excerpt": "blah blah blah",
            "fields": Object {
              "slug": "/turdsspace/",
            },
            "frontmatter": Object {
              "date": "February 03, 2018",
              "draft": false,
              "tags": Array [
                "JavaScript",
                "fp",
                "oop",
              ],
              "title": "All About JS",
            },
          }
        }
      />
      <SiteCard
        key="/turdsspace/"
        post={
          Object {
            "excerpt": "blah blah blah",
            "fields": Object {
              "slug": "/turdsspace/",
            },
            "frontmatter": Object {
              "date": "February 03, 2018",
              "draft": false,
              "tags": Array [
                "JavaScript",
                "fp",
                "oop",
              ],
              "title": "All About JS",
            },
          }
        }
      />
      <SiteCard
        key="/turdsspace/"
        post={
          Object {
            "excerpt": "blah blah blah",
            "fields": Object {
              "slug": "/turdsspace/",
            },
            "frontmatter": Object {
              "date": "February 03, 2018",
              "draft": false,
              "tags": Array [
                "JavaScript",
                "fp",
                "oop",
              ],
              "title": "All About JS",
            },
          }
        }
      />
      <SiteCard
        key="/turdsspace/"
        post={
          Object {
            "excerpt": "blah blah blah",
            "fields": Object {
              "slug": "/turdsspace/",
            },
            "frontmatter": Object {
              "date": "February 03, 2018",
              "draft": false,
              "tags": Array [
                "JavaScript",
                "fp",
                "oop",
              ],
              "title": "All About JS",
            },
          }
        }
      />
      <SiteCard
        key="/turdsspace/"
        post={
          Object {
            "excerpt": "blah blah blah",
            "fields": Object {
              "slug": "/turdsspace/",
            },
            "frontmatter": Object {
              "date": "February 03, 2018",
              "draft": false,
              "tags": Array [
                "JavaScript",
                "fp",
                "oop",
              ],
              "title": "All About JS",
            },
          }
        }
      />
      <SiteCard
        key="/turdsspace/"
        post={
          Object {
            "excerpt": "blah blah blah",
            "fields": Object {
              "slug": "/turdsspace/",
            },
            "frontmatter": Object {
              "date": "February 03, 2018",
              "draft": false,
              "tags": Array [
                "JavaScript",
                "fp",
                "oop",
              ],
              "title": "All About JS",
            },
          }
        }
      />
      <SiteCard
        key="/turdsspace/"
        post={
          Object {
            "excerpt": "blah blah blah",
            "fields": Object {
              "slug": "/turdsspace/",
            },
            "frontmatter": Object {
              "date": "February 03, 2018",
              "draft": false,
              "tags": Array [
                "JavaScript",
                "fp",
                "oop",
              ],
              "title": "All About JS",
            },
          }
        }
      />
      <SiteCard
        key="/turdsspace/"
        post={
          Object {
            "excerpt": "blah blah blah",
            "fields": Object {
              "slug": "/turdsspace/",
            },
            "frontmatter": Object {
              "date": "February 03, 2018",
              "draft": false,
              "tags": Array [
                "JavaScript",
                "fp",
                "oop",
              ],
              "title": "All About JS",
            },
          }
        }
      />
      <SiteCard
        key="/turdsspace/"
        post={
          Object {
            "excerpt": "blah blah blah",
            "fields": Object {
              "slug": "/turdsspace/",
            },
            "frontmatter": Object {
              "date": "February 03, 2018",
              "draft": false,
              "tags": Array [
                "JavaScript",
                "fp",
                "oop",
              ],
              "title": "All About JS",
            },
          }
        }
      />
      <SiteCard
        key="/turdsspace/"
        post={
          Object {
            "excerpt": "blah blah blah",
            "fields": Object {
              "slug": "/turdsspace/",
            },
            "frontmatter": Object {
              "date": "February 03, 2018",
              "draft": false,
              "tags": Array [
                "JavaScript",
                "fp",
                "oop",
              ],
              "title": "All About JS",
            },
          }
        }
      />
      <SiteCard
        key="/turdsspace/"
        post={
          Object {
            "excerpt": "blah blah blah",
            "fields": Object {
              "slug": "/turdsspace/",
            },
            "frontmatter": Object {
              "date": "February 03, 2018",
              "draft": false,
              "tags": Array [
                "JavaScript",
                "fp",
                "oop",
              ],
              "title": "All About JS",
            },
          }
        }
      />
    </List>
    <Paginate
      currentPage={1}
      totalPages={2}
    />
  </Grid>
</FlexContainer>
`;
